File: php/Pc.php
<?php 

class Pc{

    // deklarasi dan inisialisasi atribut private termasuk composition dengan objek Processor, Disk, dan Ram
    private Processor $Pobj;
    private Disk $Dobj;
    private Ram $Robj;
    private $totalPrice = 0;


    // constructor dengan parameter
    public function __construct($Pobj = Processor, $Dobj = Disk, $Robj = Ram){
        $this->Pobj = $Pobj;
        $this->Dobj = $Dobj;
        $this->Robj = $Robj;
    }


    // deklarasi setter sebagai method (Write)
    public function setP($Pobj){
        $this->Pobj = $freq;
    }

    public function setD($Dobj){
        $this->Dobj = $size;
    }

    public function setR($Robj){
        $this->Robj = $supCuda;
    }

    // setter pada total price adalah penjumlahan terhadap ketiga harga pada kelas spek PC
    public function setTotalPrice($priceP = 0, $priceD = 0, $priceR = 0){
        $this->totalPrice = $priceP + $priceD  + $priceR;
    }

    
    // deklarasi getter sebagai method (Read Only)
    public function getP(){
        return $this->Pobj;
    }

    public function getD(){
        return $this->Dobj;
    }

    public function getR(){
        return $this->Robj;
    }

    public function getTotal(){
        return $this->totalPrice;
    }

    
    // deklarasi method Display untuk menampilkan data menggunakan method Getter
    public function countTotal(){
        echo "Total Price    = $". $this->getTotal(). "<br>";
    }


    // destructor
    public function __destruct(){

    }
        
        
}

?>

File: java/Pc.java
public class Pc {
    
    // deklarasi dan inisialisasi atribut private termasuk composition dengan objek Processor, Disk, dan Ram
    private Processor Pobj;
    private Disk Dobj;
    private Ram Robj;
    int totalPrice = 0;


    // constructor tanpa parameter
    public void Pc(){
        
    }

    // constructor dengan parameter
    public void Pc(Processor inputP, Disk inputD, Ram inputR, int total){
        this.Pobj = inputP;
        this.Dobj = inputD;
        this.Robj = inputR;

        // jumlahkan seluruh dengan memanggil masing-masing Getter price kemudian ditampung dalam atribut totalPrice
        this.totalPrice = Pobj.getPriceP() + Dobj.getPriceD() + Robj.getPriceR();
    }

    
    // deklarasi setter sebagai method (Write)
    public void setP(Processor P){
        this.Pobj = P;
    }

    public void setD(Disk D){
        this.Dobj = D;
    }

    public void setR(Ram R){
        this.Robj = R;
    }

    public void setTotal(int priceP, int priceD, int priceR){
        this.totalPrice = priceP + priceD + priceR;
    }


    // deklarasi getter sebagai method (Read Only)
    public Processor getP(){
        return this.Pobj;
    }

    public Disk getD(){
        return this.Dobj;
    }

    public Ram getR(){
        return this.Robj;
    }

    public int getTotal(){
        return this.totalPrice;
    }


    // deklarasi method Display untuk menampilkan data menggunakan method Getter
    public void countTotal(){
        System.out.println("Total Price    = $" + this.getTotal());
    }
}
